### ISSUE TITLE
Infinite loop occurs in DexPrinter.findLongJumps() while processing android framework

### ISSUE THREAD

**Issue by madberries** [Sable/soot#486]
There appears to be an infinite loop in ```soot.toDex.DexPrinter.findLongJumps()``` while processing ```android.text.StaticLayout.generate(java.lang.CharSequence,int,int,android.text.TextPaint,int,android.text.TextDirectionHeuristic,float,float,boolean,boolean,float,android.text.TextUtils$TruncateAt)``` in the android framework.  You can reproduce this by supplying the dex file at http://people.csail.mit.edu/jeikenberry/classes2.dex to the following main class:

```
public class Main {
    public static void main(String[] args) {
		
        //prefer Android APK files// -src-prec apk
        Options.v().set_src_prec(Options.src_prec_apk);
		
        //output as APK, too//-f J
        Options.v().set_output_format(Options.output_format_dex);        
        
        Options.v().set_whole_program(true);
           
        PhaseOptions.v().setPhaseOption("cg", "off");
        
        Options.v().set_allow_phantom_refs(true);            
         
        soot.Main.main(args);       
    }       
}
```
Note: the infinite loop was introduced on commit 85e66a844218134d5ffb18f3b5f1349a465a4c94.

**Comment by Alexandre-Bartel** [comment]
Thank you for reporting this. It should have been fixed by cc7306dcb2acd95d3867c7a012412dd2462d20d0. Please confirm and then close the issue.

**Comment by madberries** [comment]
This fix appears to work, thanks!
